### ISSUE TITLE
categorical variables do not work with skip_learning

### ISSUE THREAD

**Issue by raphaelhoffmann** [HazyResearch/deepdive#463]
When using `skip_learning` and loading a weight_table, the weights are assigned by joining only on the  `description` column but not the `cardinality` column. See the code [here](https://github.com/HazyResearch/deepdive/blob/e770f7e9ea7ff71dedc05e6392349f459807bbfd/src/main/scala/org/deepdive/inference/SQLInferenceRunner.scala#L913).

Consequently, the same feature weights are assigned to every category, eg.:
```
  3958 |       0 | -0.0466067 | 00003       | factor_feature-^_^
  3957 |       0 | -0.0466067 | 00002       | factor_feature-^_^
  3956 |       0 | -0.0466067 | 00001       | factor_feature-^_^
  3955 |       0 | -0.0466067 | 00000       | factor_feature-^_^
  3959 |       0 | -0.0466067 | 00004       | factor_feature-^_^
 37944 |       0 |   -1.91256 | 00004       | factor_feature-¨
 37940 |       0 |   -1.91256 | 00000       | factor_feature-¨
 37943 |       0 |   -1.91256 | 00003       | factor_feature-¨
 37942 |       0 |   -1.91256 | 00002       | factor_feature-¨
 37941 |       0 |   -1.91256 | 00001       | factor_feature-¨
``` 
This is wrong; the fix is to consider both cardinality and description when loading weights.

@zifeishan and I are fixing this in our branch for now. Does this issue also exist in DD0.8?

**Comment by netj** [comment]
DD0.8 currently does not support learned weight reusing yet, so no it doesn't have the issue now but should be careful.

**Comment by chrismre** [comment]
Ugh! I think we need this feature of reusing weights... This has got to be in a quick fix for 0.8.1 or something... @raphaelhoffmann does your fix work on 0.8? Can you create a PR for it?

**Comment by netj** [comment]
Yes this is super important and I'm trying to come up with the right way to do this hopefully before this release. @raphaelhoffmann and I synced up on this. The fix to Scala code base won't be portable to 0.8.
